services/src/main/java/com/chatbot/services/MessageSenders/HangoutsMessageSender.java

```diff
@@ -0,0 +1,100 @@
+  private static String CHAT_SCOPE;
+  private static GoogleCredentials credentials;
+  private static HttpRequestInitializer requestInitializer;
+  private static HangoutsChat chatService;
+
+  HangoutsMessageSender(@Value("${hangoutsAPIScope}") final String apiScope,
+      @Value("${credentialsFile}") final String credentialsFile) throws GeneralSecurityException,
+      IOException {
+    CHAT_SCOPE = apiScope;
+    credentials = GoogleCredentials.fromStream(
+        HangoutsMessageSender.class.getResourceAsStream(credentialsFile))
+        .createScoped(CHAT_SCOPE);
+    requestInitializer = new HttpCredentialsAdapter(credentials);
+    chatService = new HangoutsChat.Builder(
+        GoogleNetHttpTransport.newTrustedTransport(),
+        JacksonFactory.getDefaultInstance(), requestInitializer)
+        .setApplicationName("chatbot").build();
+  }
+
+  // send message to a user using their chatClientGeneratedID (spaceID for hangouts)
+  @Override
+  public void sendMessage(final String chatClientGeneratedID, final String msg) throws IOException {
+    if(msg.isEmpty()) {
+      return;
+    }
+    final Message message = new Message().setText(msg);
+    chatService.spaces()
+        .messages()
+        .create("spaces/" + chatClientGeneratedID, message)
+        .execute();
+  }
+
+  // send an interactive card message to a user using their chatClientGeneratedID
+  public void sendCardMessage(final String chatClientGeneratedID, final String msg)
+      throws IOException {
+    // responses to be sent as cards are newline seperated, the first line is the description and
+    // the following lines are options to be shown
+    final List<String> messageParts = new ArrayList<>(Arrays.asList(msg.split("\n")));
+    final List<Section> sectionList = new ArrayList<>();
+    for (final String option : messageParts.subList(1, messageParts.size())) {
+      final List<WidgetMarkup> widgets = new ArrayList<>();
+      final List<ActionParameter> customParameters = Collections
+          .singletonList(new ActionParameter()
+          .setKey("message")
+          .setValue(option));
+      final FormAction action = new FormAction()
+          .setActionMethodName("INTERACTIVE_TEXT_BUTTON_ACTION")
+          .setParameters(customParameters);
+      final OnClick onClick = new OnClick().setAction(action);
+      final TextButton button = new TextButton()
+          .setText(option)
+          .setOnClick(onClick);
+      final Button widget = new Button().setTextButton(button);
+      widgets.add(new WidgetMarkup().setButtons(Collections.singletonList((widget))));
+      final Section section = new Section().setWidgets(widgets);
+      sectionList.add(section);
+    }
+    final Card card =
+        (new Card()).setSections(Collections.unmodifiableList(sectionList));
+    final Message message = new Message().setCards(Collections.singletonList(card));
+    // the description is sent first followed by the options
```

Comment:
create a private foo for generating card message and test that.

Reply:
updated